lib/ts/recipe/emailverification/components/features/emailVerification/index.tsx

```diff
@@ -85,10 +85,9 @@ class EmailVerification extends PureComponent<Prop, { status: "READY" | "LOADING
             }
 
             // we check if the email is already verified, and if it is, then we redirect the user
-            // TODO NEMI: handle user context for pre built UI
             const isVerified: boolean = (
                 await this.props.recipe.recipeImpl.isEmailVerified({
-                    userContext: {},
+                    userContext: getNormalisedUserContext(this.props.userContext),
```

Comment:
It's strange that in some places you need to use `getNormalisedUserContext`, and in others, you don't.

Reply:
I would suggest that you make all components that are exposed to the user be wrapped around another HOC whose only job is to accept an optional userContext prop and use the userContextProvider component

Reply:
Done